types/lodash/common/collection.d.ts

```diff
@@ -1781,6 +1781,10 @@ declare module "../index" {
             collection: List<T> | null | undefined,
             callback: ValueIteratee<T>
         ): [T[], T[]];
+        partition<T1 | T2>(
+            collection: List<T1 | T2> | null | undefined,
+            callback: ValueIteratee<T1 | T2>
```

Comment:
I think this should be something with a type guard, maybe:

`ValueIteratorTypeGuard<T1 | T2, T1>`

Or basically:

`(value: T1 | T2) => value is T1`

Also, this new overload should probably be the first overload. Otherwise TS will always choose the old overload, unless you explicitly specify the type parameters.

Reply:
Will try this